tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -223,6 +223,50 @@ ROOT::Experimental::Detail::RPageSinkDaos::CommitSealedPageImpl(DescriptorId_t c
    return result;
 }
 
+std::vector<ROOT::Experimental::RNTupleLocator>
+ROOT::Experimental::Detail::RPageSinkDaos::CommitSealedPageVImpl(std::span<RPageStorage::RSealedPageGroup> ranges)
+{
+   RDaosContainer::MultiObjectRWOperation_t writeRequests;
+   std::vector<ROOT::Experimental::RNTupleLocator> locators;
+   locators.reserve(ranges.size());
+
+   DescriptorId_t clusterId = fDescriptorBuilder.GetDescriptor().GetNClusters();
+   std::size_t szPayload = 0;
+   unsigned numPages = 0;
+
+   /// Aggregate batch of requests by object ID and distribution key, determined by the ntuple-DAOS mapping
+   for (auto &range : ranges) {
+      for (auto sealedPageIt = range.fFirst; sealedPageIt != range.fLast; ++sealedPageIt) {
+         const RPageStorage::RSealedPage &s = *sealedPageIt;
+         d_iov_t pageIov;
+         d_iov_set(&pageIov, const_cast<void *>(s.fBuffer), s.fSize);
+         auto offsetData = fPageId.fetch_add(1);
+
+         RDaosKey daosKey = GetPageDaosKey<kDefaultDaosMapping>(clusterId, range.fColumnId, offsetData);
+         auto odPair = RDaosContainer::ROidDkeyPair{daosKey.fOid, daosKey.fDkey};
+         auto [it, ret] = writeRequests.emplace(odPair, RDaosContainer::RWOperation(odPair));
+         it->second.insert(daosKey.fAkey, pageIov);
+
+         RNTupleLocator result;
+         result.fPosition = offsetData;
+         result.fBytesOnStorage = s.fSize;
+         locators.push_back(result);
+
+         szPayload += s.fSize;
+         ++numPages;
+      }
+   }
+   fNBytesCurrentCluster += szPayload;
+
+   RNTupleAtomicTimer timer(fCounters->fTimeWallWrite, fCounters->fTimeCpuWrite);
+   if (int err = fDaosContainer->WriteV(writeRequests))
+      throw ROOT::Experimental::RException(R__FAIL("WriteV: error" + std::string(d_errstr(err))));
```

Comment:
I would probably create a compound statement, i.e. `{ ... }` around this, so that `timer` is scoped and we don't accidentally measure anything below.